### PR TITLE
chore(dynamite_runtime): fix last pana complaints for the 0.1.0 release

### PR DESCRIPTION
**Pull request by Leptopoda**
I'm not sure whether the repository url should really point to the root repo.
both json_serializable and built_value (to name two larger monorepos) point the repo url to the directory the package lives in:
https://github.com/google/json_serializable.dart/blob/master/json_serializable/pubspec.yaml
https://github.com/google/built_value.dart/blob/master/built_value/pubspec.yaml

Attached are the pana results but the only remaining issue is that pana can't find the package in the clone of the root repository :woman_shrugging: 
[dynamite_pana.txt](https://github.com/nextcloud/neon/files/13758807/dynamite_pana.txt)
[dynamite_runtime_pana.txt](https://github.com/nextcloud/neon/files/13758808/dynamite_runtime_pana.txt)

### CODE REVIEW

File: packages/dynamite/dynamite/CHANGELOG.md
Comment (provokateurin) [inline packages/dynamite/dynamite/CHANGELOG.md:1]: Don't add this, melos version will generate it

File: packages/dynamite/dynamite/pubspec.yaml [hunk 1]
```diff
@@ -1,5 +1,13 @@
 name: dynamite
-version: 1.0.0
+version: 0.1.0
```
Comment (provokateurin) [inline packages/dynamite/dynamite/pubspec.yaml:2]:
```suggestion
version: 0.1.0-dev.0
```

Needed so melos version will "upgrade" it to 0.1.0

File: packages/dynamite/dynamite_runtime/CHANGELOG.md
Comment (provokateurin) [inline packages/dynamite/dynamite_runtime/CHANGELOG.md:1]: Also remove this

**Comment by Leptopoda** [comment]
Ah and we still can't automatically add the petstore example. For now I just copied the contents to the example directories and adjusted the pubspec_overrides. Long term we should have CI/CD  for that.

File: packages/dynamite/dynamite_runtime/pubspec.yaml [hunk 2]
```diff
@@ -1,6 +1,13 @@
 name: dynamite_runtime
-description: Helper package for use with dynamite.
-version: 1.0.0
+description: The helper package used at runtime by code generated with the OpenAPI generator dynamite.
+version: 0.1.0
```
Comment (provokateurin) [inline packages/dynamite/dynamite_runtime/pubspec.yaml:3]: Again as prerelease please

**Review by provokateurin — CHANGES_REQUESTED** [review]

File: packages/dynamite/dynamite/CHANGELOG.md
Comment (Leptopoda) [reply packages/dynamite/dynamite/CHANGELOG.md:1]: I'm not that sure.
I don't think the initial release should contain all the commits so far :thinking:

**Comment by provokateurin** [comment]
pana complains that web is not supported in dynamite_runtime, but it is because we are using universal_io. Please add this to the pubspec:
```yaml
platforms:
  windows:
  linux:
  macos:
  android:
  ios:
  web:
```
For dynamite configuring the platforms doesn't really make sense, but we could limit it to linux, macos and windows

**Comment by provokateurin** [comment]
> Ah and we still can't automatically add the petstore example. For now I just copied the contents to the example directories and adjusted the pubspec_overrides. Long term we should have CI/CD for that.

Even though it's stupid, we could just always copy the package to the example folders when generating it. Then we have 3 copies in the repo, but I feel like that is the easiest way.

We could also just make it the example package of dynamite. For dynamite_runtime we don't really need to care about the score. This would be cleaner imo

**Comment by Leptopoda** [comment]
> We could also just make it the example package of dynamite.

I'd rather do that and drop the other packages.
The runtime can just have a README with a link.

**Comment by provokateurin** [comment]
> I'm not sure whether the repository url should really point to the root repo.

Yeah please point it to the path as well.

**Comment by Leptopoda** [comment]
Why not just switch it around so the homepage points to the root?
I don't think `homepage` and `repository` should point to the same.

**Comment by provokateurin** [comment]
Pana will try to check for the package using the repository url. I'm not sure how it is supposed to be used, so please just do what pana wants us to do.

**Comment by Leptopoda** [comment]
Ok cspell needs adjusting but review is appreciated.
I don't think we can make pana happy with the repo url but at least it's now aligned with how built_value and json_serializable do it.

**Comment by Leptopoda** [comment]
any news here?
I think dynamite might be ready :)

**Comment by provokateurin** [comment]
please rebase

**Comment by provokateurin** [comment]
and run pana again

**Comment by Leptopoda** [comment]
Pana still complains about the pubspec and the missing changelog.

File: packages/dynamite/dynamite/pubspec.yaml [hunk 3]
```diff
@@ -1,5 +1,13 @@
 name: dynamite
-version: 1.0.0
+version: 0.1.0-dev.0
+description: The helper package used at runtime by code generated with the OpenAPI generator dynamite.
+homepage: https://github.com/nextcloud/neon
+repository: https://github.com/nextcloud/neon/tree/main/packages/dynamite/dynamite
+issue_tracker: https://github.com/nextcloud/neon/issues
```
Comment (provokateurin) [inline packages/dynamite/dynamite/pubspec.yaml:6]: The issue tracker is not needed when the repository points to GH (https://dart.dev/tools/pub/pubspec#issue-tracker)

File: packages/dynamite/dynamite_runtime/pubspec.yaml [hunk 4]
```diff
@@ -1,6 +1,13 @@
 name: dynamite_runtime
-description: Helper package for use with dynamite.
-version: 1.0.0
+description: The helper package used at runtime by code generated with the OpenAPI generator dynamite.
+version: 0.1.0-dev.0
+homepage: https://github.com/nextcloud/neon
+repository: https://github.com/nextcloud/neon/tree/main/packages/dynamite/dynamite_runtime
+issue_tracker: https://github.com/nextcloud/neon/issues
```
Comment (provokateurin) [inline packages/dynamite/dynamite_runtime/pubspec.yaml:6]: Same as above

**Review by provokateurin — APPROVED** [review]

File: packages/dynamite/dynamite_runtime/pubspec.yaml [hunk 5]
```diff
@@ -22,3 +29,11 @@ dev_dependencies:
...
       path: packages/neon_lints
   test: ^1.24.9
+
+platforms:
+  windows:
+  linux:
+  macos:
+  android:
+  ios:
+  web:
```
Comment (provokateurin) [inline packages/dynamite/dynamite_runtime/pubspec.yaml:39]: I think this should be further up, before or after the environment

File: melos.yaml [hunk 6]
```diff
@@ -13,6 +13,7 @@ packages:
   - packages/*
   - packages/dynamite/*
   - packages/neon/*
+  - packages/**/example
```
Comment (provokateurin) [inline melos.yaml:16]: Not needed as example packages are automatically included

**Review by provokateurin — CHANGES_REQUESTED** [review]

File: packages/dynamite/dynamite_runtime/pubspec.yaml (hunk 5 above)
Comment (Leptopoda) [reply packages/dynamite/dynamite_runtime/pubspec.yaml:39]: I prefer not to scroll to get to the relevant information (deps and environment).
But am open if you want it further up.

File: melos.yaml (hunk 6 above)
Comment (Leptopoda) [reply melos.yaml:16]: melos will not generate a pubspec_override file if examples are not included.

This means we either need use hard coded paths in the pubspec or use productions releases.

File: packages/dynamite/dynamite_runtime/pubspec.yaml (hunk 5 above)
Comment (provokateurin) [reply packages/dynamite/dynamite_runtime/pubspec.yaml:39]: Good point

File: melos.yaml (hunk 6 above)
Comment (provokateurin) [reply melos.yaml:16]: Ah I only checked if the pubspec.lock is regenerated.

**Comment by provokateurin** [comment]
Can you rebase and push the issue tracker fix? Then this is ready to be merged

**Review by provokateurin — APPROVED** [review]

**Pull request merged**